packages/shared_preferences/shared_preferences_tool/lib/src/ui/keys_panel.dart

```diff
@@ -0,0 +1,233 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app_shared/ui.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+import '../async_state.dart';
+import '../shared_preferences_state.dart';
+import '../shared_preferences_state_notifier.dart';
+import '../shared_preferences_state_notifier_provider.dart';
+import 'error_panel.dart';
+
+/// A panel that displays the keys stored in shared preferences.
+class KeysPanel extends StatefulWidget {
+  /// Default constructor for [KeysPanel].
+  const KeysPanel({super.key});
+
+  @override
+  State<KeysPanel> createState() => _KeysPanelState();
+}
+
+class _KeysPanelState extends State<KeysPanel> {
+  bool searching = false;
+  final FocusNode searchFocusNode = FocusNode();
+
+  @override
+  void dispose() {
+    searchFocusNode.dispose();
+    super.dispose();
+  }
+
+  void _startSearching() {
+    setState(() {
+      searching = true;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final SharedPreferencesStateNotifier notifier =
+        SharedPreferencesStateNotifierProvider.of(context);
+
+    void stopSearching() {
+      setState(() {
+        searching = false;
+      });
+      notifier.filter('');
+    }
+
+    return RoundedOutlinedBorder(
+      clip: true,
+      child: Column(
+        children: <Widget>[
+          AreaPaneHeader(
+            roundedTopBorder: false,
+            includeTopBorder: false,
+            tall: true,
+            title: Row(
+              children: <Widget>[
+                Text(
+                  'Stored Keys',
+                  style: Theme.of(context).textTheme.titleSmall,
+                ),
+                if (searching) ...<Widget>[
+                  const SizedBox(
+                    width: denseSpacing,
+                  ),
+                  Expanded(
+                    child: KeyboardListener(
+                      focusNode: searchFocusNode,
+                      onKeyEvent: (KeyEvent value) {
+                        if (value.logicalKey == LogicalKeyboardKey.escape) {
+                          stopSearching();
+                        }
+                      },
+                      child: TextField(
+                        autofocus: true,
+                        decoration: InputDecoration(
+                          contentPadding: const EdgeInsets.symmetric(
+                            horizontal: densePadding,
+                          ),
+                          hintText: 'Search',
+                          border: const OutlineInputBorder(),
+                          suffix: Tooltip(
+                            message: 'Stop searching',
+                            child: IconButton(
```

Comment:
same comment with respect to buttons across this PR. Using `DevToolsButton` is recommended.